src/gui/WndMain.cpp

```diff
@@ -65,9 +65,36 @@
 
 static int gameLogoWidth, gameLogoHeight;
 static int splashLogoWidth, splashLogoHeight;
+
 static HANDLE hPersistedMemory = NULL;
 static LPVOID PersistedMemoryAddr = nullptr;
 
+void MapPersistedMemory()
+{
+	assert((hPersistedMemory == NULL) == (PersistedMemoryAddr == nullptr));
+
+	if (hPersistedMemory == NULL) {
+		hPersistedMemory = OpenFileMapping(FILE_MAP_READ, FALSE, "PersistentMemory");
+		assert(hPersistedMemory != NULL);
+
+		PersistedMemoryAddr = MapViewOfFile(hPersistedMemory, FILE_MAP_READ, 0, 0, 0);
+		assert(PersistedMemoryAddr != nullptr);
+	}
+}
+
+void UnmapPersistedMemory()
+{
+	assert((hPersistedMemory == NULL) == (PersistedMemoryAddr == nullptr));

```

Comment:
Upon a quick reboot, `hPersistedMemory` and `PersistedMemoryAddr` will both be non-null (because they were assigned in the previous instance of cxbxr), and thus the equality becomes `false == false` -> true, so the assertion is not triggered. However, in my opinion, it's still misleading becasuse it would suggest that both globals are expected to be null, when in reality they will be non-null. So I would rewrite this as `assert((hPersistedMemory != NULL) == (PersistedMemoryAddr != nullptr));`

Reply:
You're right, that does look fishy like this - I quickly copied over the functionally equivalent check from the mapping function, but checking both for non-null would indeed be more accurate

Reply:
Updated with your suggestion